include/sas.h

```diff
@@ -114,42 +110,46 @@ class SAS
   typedef uint64_t TrailId;
   typedef uint64_t Timestamp;
 
-#if HAVE_ZLIB_H
-  // Compression-related classes are only available if zlib is
+  // Compression profile
   class Profile
   {
   public:
-    inline Profile(std::string dictionary) : _dictionary(dictionary) {}
+    enum CompressionType {
+      ZLIB = 0,
+      LZ4
+    };
+
+    inline Profile(std::string dictionary, CompressionType t = ZLIB) : _dictionary(dictionary), _type(t) {}
+    inline Profile(CompressionType t) : _dictionary(""), _type(t) {}
     inline const std::string& get_dictionary() const {return _dictionary;}
+    inline bool is_lz4() const {return _type == LZ4;}
 
   private:
     const std::string _dictionary;
+    const CompressionType _type;
   };
 
   class Compressor
   {
   public:
-    static Compressor* get();
-
-    std::string compress(const std::string& s, const Profile* profile = NULL);
+    virtual std::string compress(const std::string& s, std::string dictionary) = 0;
+    static Compressor* get(const Profile* profile);
```

Comment:
I'd prefer not to have CompressionType/Algorithm be nested within Compressor, just because I think the API user currently never has to see Compressor at the moment, and this would change that.

I'm happy to just pass in the `SAS::Profile::Algorithm` here, though.